### PR TITLE
feat: Implement Github CLI "pr merge" command

### PR DESCRIPTION
**Pull request by Shmarkus**
Implemented ```gh pr merge``` with the possibility to define the branch or additional parameters

### CODE REVIEW

**Comment by KyleTryon** [comment]
Thank you for reminding me the `CHANGELOG` is still present here. Happy to take the change here, but we no longer user the changelog. I'll be sure to remove it in a change soon. We now use GitHub's Releases API. 

https://github.com/CircleCI-Public/github-cli-orb/releases

When we publish this change, we will use this PR to populate the release notes. I'll be sure to update all that documentation. Thank you.

The code looks good. Could I ask how you intend to use this @Shmarkus ? Is there an example of config you could provide on how you plan to execute this?

By default CircleCI will trigger on a Push (including a Tag), we can augment this a bit more with https://github.com/CircleCI-Public/trigger-circleci-pipeline-action

In what scenario would you run this job?

**Comment by Shmarkus** [comment]
I can create a concrete example as well but since I'm not at the computer at the moment, I'll just illustrate my use-case.

At the moment, the Dependabot is configured to check for version updates. It creates a pull request with the new version. (Renovate bot has the same functionality)

When the test pipeline succeeds I want to merge the PR automatically (this is where the pain is at the moment)

**Comment by Shmarkus** [comment]
Added example, removed changelog from changelog

**Review by KyleTryon — APPROVED** [review]
Looks excellent and the example is a great/creative use-case, thank you!

File: src/examples/pr-merge.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+description: >
+  Use the GitHub CLI merge pull request automatically. Usually it can be a pull request from a bot like Dependabot.
+  In this example, on every Dependabot pull request, we run tests and merge the PR if tests pass 
+  Add a Context containing your GITHUB_TOKEN and GITHUB_HOSTNAME (optional) or set a project-level environment variable.
+usage:
+  version: 2.1
+  orbs:
+    gh: circleci/github-cli@2.2.1
```
Comment (KyleTryon) [inline src/examples/pr-merge.yml:8]:
```suggestion
    gh: circleci/github-cli@2.2.0
```

This will be a new minor with no patches

**Pull request merged**